File: Functions/update_budget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, expenses, spend, total; 

float input(char type[], float var){
    printf("How much is your %s:  \n", type); 
    scanf( "%f" , &var); 
    return var;
}


void percent(char type[], int amount){
    int per = amount/income *100;
    printf("Your %s is %d%% of your income.\n", type, per); 
}

int main(void) {
    printf("This is going to calculate your budget for the month.");
    income = input("income", income);
    rent = input("rent", rent);
    utilities = input("utilities", utilities);
    groceries = input("groceries", groceries); 
    transportation = input("transportation", transportation); 
    savings = income *.2;
    expenses = rent + utilities + groceries + transportation; 
    spend = income - expenses - savings;
    percent ("rent", rent);
    printf("Your expenses are: $%.2f%%", expenses);
    printf("Your savings are: $%.2f%%", savings); 
    printf("Your total left to spend is: $%.2f\n", total);

     percent ("rent", rent);
     percent ("utilities", utilities);
     percent ("groceries", groceries); 
     percent ("transportation", transportation); 
     percent ("savings", savings);
     percent ("expenses", expenses); 

    return 0;
}

File: Functions/update_budget.py

print("This is going to calculate your budget for the month.")
income = float (input("What is your monthly income: "))
rent = float (input("What is your monthly rent: "))
utilities = float (input("What is your monthly utilities: "))
groceries = float (input("What is your monthly groceries: "))
transportation = float (input("What is your monthly transportation: "))
savings = income *.2
expenses = rent + utilities + groceries + transportation
total = income - expenses - savings
prent = rent/income * 100
putilities = utilities/income *100
pgroceries = groceries/income *100
ptransportation = transportation/income * 100
psavings = savings/income *100 
pexpenses = expenses/income * 100
def percent(type, amount):
    per = amount/income *100

    return (f"Your {type} is {per}% income. ")


print(f"Your monthly income is ${income}\n")
print(f"Your monthly expenses are ${expenses}\n")
print(f"Your monthly savings is ${savings}\n")
print(percent("rent", rent))
print(percent("utilities", utilities))
print(percent("groceries", groceries))
print(percent("transportation", transportation))
print(percent("savings", savings))
print(percent("expenses", expenses))